backend/lib/deal-observer.js

```diff
@@ -16,9 +15,10 @@ import { Value } from '@sinclair/typebox/value'
 export async function observeBuiltinActorEvents (blockHeight, pgPool, makeRpcRequest) {
   const eventType = 'claim'
   const activeDeals = await getActorEvents(getActorEventsFilter(blockHeight, eventType), makeRpcRequest)
-  assert(activeDeals !== undefined, `No ${eventType} events found in block ${blockHeight}`)
-  console.log(`Observed ${activeDeals.length} ${eventType} events in block ${blockHeight}`)
-  await storeActiveDeals(activeDeals, pgPool)
+  if (activeDeals.length > 0) {
+    console.log(`Observed ${activeDeals.length} ${eventType} events in block ${blockHeight}`)
+    await storeActiveDeals(activeDeals, pgPool)
+  }
```

Comment:
I find it important for the logs to tell us when we successfully fetched & processed actor events, but there were no deals found, as a confirmation that the process finished.

I also think the short-circuit handling of empty list of active deals is better handled inside `storeActiveDeals` function. 1) Every place calling `storeActiveDeals()` benefit from it. 2) It's easier to write a test to verify this behaviour.

Reply:
Here is what I would do myself:

```diff
 export async function storeActiveDeals (activeDeals, pgPool) {
+  if (!activeDeals.length) return

   const transformedDeals = activeDeals.map((deal) => (
```